components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/LegacyJobOrchestrator.cs

```diff
@@ -522,87 +581,54 @@ public class LegacyJobOrchestrator : IJobOrchestrator, IPublisherConfigServices,
+                    request.OpcNodes = nodesToRemoveSet.ToList();
+                    var entriesNotFoundJson = _jsonSerializer.SerializeToString(request);
+                    throw new MethodCallStatusException(entriesNotFoundJson, (int)HttpStatusCode.NotFound, "Nodes not found");
                 }
 
-                foreach (var entry in existingGroup) {
-                    if (!entry.OpcNodes.Any()) {
-                        _publishedNodesEntrys.Remove(entry);
-                    }
-                }
+                updatedPublishedNodesEntries = updatedPublishedNodesEntries.Where(entry => entry.OpcNodes.Count > 0).ToList();
 
-                var found = false;
-                await _lockJobs.WaitAsync(ct).ConfigureAwait(false);
-                try {
-                    var jobs = _publishedNodesJobConverter.ToWriterGroupJobs(existingGroup, _legacyCliModel);
-                    if (jobs.Any()) {
-                        foreach (var job in jobs) {
-                            var newJob = ToJobProcessingInstructionModel(job);
-                            if (string.IsNullOrEmpty(newJob?.Job?.Id)) {
-                                continue;
-                            }
-                            foreach (var assignedJob in _assignedJobs) {
-                                if (newJob.Job.Id == assignedJob.Value.Job.Id) {
-                                    _assignedJobs[assignedJob.Key] = newJob;
-                                    found = true;
-                                    break;
-                                }
-                            }
-                            if (!found) {
-                                if (_availableJobs.ContainsKey(newJob.Job.Id)) {
-                                    _availableJobs[newJob.Job.Id] = newJob;
-                                    found = true;
-                                }
-                            }
-                        }
-                    }
+                _publishedNodesEntries.Clear();
+                _publishedNodesEntries.AddRange(updatedPublishedNodesEntries);
 
-                    if (!found) {
-                        var entryJobId = _publishedNodesJobConverter.
-                            ToConnectionModel(request, _legacyCliModel).CreateConnectionId();
-                        foreach (var assignedJob in _assignedJobs) {
-                            if (entryJobId == assignedJob.Value.Job.Id) {
-                                found = _assignedJobs.Remove(assignedJob.Key, out _);
-                                break;
-                            }
-                        }
-                        if (!found) {
-                            found = _availableJobs.Remove(entryJobId, out _);
-                        }
-                    }
-                }
-                finally {
-                    _lockJobs.Release();
-                }
+                PersistPublishedNodes();
```

Comment:
>PersistPublishedNodes [](http://example.com/codeflow?start=16&length=21)

probably we should call persist after we do refresh jobs since this will change the _publishedNodesEntries once again

Reply:
I took update of `_publishedNodesEntries` out of `RefreshJobs()` method. Now `RefreshJobs()` only updates `_availableJobs` and `_assignedJobs`.